lib/sensors/wheel_encoder.cpp

```diff
@@ -0,0 +1,36 @@
+#include "wheel_encoder.hpp"
+
+namespace hyped::sensors {
+
+WheelEncoder::WheelEncoder(core::ILogger &logger, std::shared_ptr<io::IAdc> adc)
+    : logger_(logger),
+      adc_(adc),
+      count_(0),
+      previous_voltage_(0)
+{
+}
+
+std::uint64_t WheelEncoder::getCount()
+{
+  return count_;
+}
+
+core::Result WheelEncoder::updateCount()
+{
+  const auto optional_voltage = adc_->readValue();
+  if (!optional_voltage) {
+    logger_.log(core::LogLevel::kFatal, "Failed to read wheel encoder value from ADC");
+    return core::Result::kFailure;
+  }
+  const core::Float voltage = *optional_voltage;
+  if (voltage > kVoltageThreshold && previous_voltage_ < kVoltageThreshold) { count_++; }
+  previous_voltage_ = voltage;
+  return core::Result::kSuccess;
```

Comment:
If we keep getting a value over the thresh, at what point do we think the encoders aren't working. Ik nav pays a lot of attention to these so want to make sure we report any possible failure asap.

Reply:
Yeah these are our singular most accurate sensor I think. If these give faulty data, we'll only spot it if it's disagreeing with our other data but this data being unreliable could potentially leave us without a ground truth

Reply:
it depends on velocity. we can probably calculate a maximum high period before we stop trusting it based off of minimum velocity we expect to hold